www/class/centreonHost.class.php

```diff
@@ -1312,6 +1383,34 @@ public function getTemplateChain(
         return $templates;
     }
 
+    /**
+     * @param $hostId
+     * @param $alreadyProcessed
+     * @return array
+     * @throws Exception
+     */
+    public function getHostChain(
```

Comment:
Why return an array if you use its reference to update it?

Reply:
to get the data from the n-th level